include/oi/result/Element.h

```diff
@@ -40,6 +42,7 @@ struct Element {
   size_t exclusive_size;
 
   std::optional<uintptr_t> pointer;
+  std::variant<std::nullopt_t, uint64_t, std::string> data = {std::nullopt};
```

Comment:
It would make more sense to me to disambiguate the pointer/general number entry here. You can do:
```
struct Pointer { uintptr_t p; };
struct Scalar { uint64_t s; };
std::variant<std::nullopt_t, Pointer, Scalar, std::string> data;
```
Which leaves the interface still quite clean.

This would require changing the non-specialised capture function to a `Sum` though. Your choice.